### ISSUE TITLE
3D plot with matplotlib error

### ISSUE THREAD

**Issue by pinkardw** [sagemath/cloud#196]
The following code (taken directly from the matplotlib 3D tutorial) does not work on the cloud:

    import matplotlib.pyplot as plt
    from mpl_toolkits.mplot3d import Axes3D
    fig = plt.figure()
    ax = fig.add_subplot(111, projection='3d')
    fig.savefig('test.png')

It does work on my local binary of Sage 6.6. I get the following error from the cloud:

```
Traceback (most recent call last):
  File "/projects/c7ac720e-5eb9-4f38-988a-1dbd4628dca9/.sagemathcloud/sage_server.py", line 879, in execute
    exec compile(block+'\n', '', 'single') in namespace, locals
  File "", line 1, in <module>
  File "/usr/local/sage/sage-6.5/local/lib/python2.7/site-packages/matplotlib-1.3.1-py2.7-linux-x86_64.egg/matplotlib/figure.py", line 1221, in gca
    return self.add_subplot(1, 1, 1, **kwargs)
  File "/usr/local/sage/sage-6.5/local/lib/python2.7/site-packages/matplotlib-1.3.1-py2.7-linux-x86_64.egg/matplotlib/figure.py", line 914, in add_subplot
    a = subplot_class_factory(projection_class)(self, *args, **kwargs)
  File "/usr/local/sage/sage-6.5/local/lib/python2.7/site-packages/matplotlib-1.3.1-py2.7-linux-x86_64.egg/matplotlib/axes.py", line 9259, in __init__
    self._axes_class.__init__(self, fig, self.figbox, **kwargs)
  File "/usr/lib/python2.7/dist-packages/mpl_toolkits/mplot3d/axes3d.py", line 97, in __init__
    *args, **kwargs)
  File "/usr/local/sage/sage-6.5/local/lib/python2.7/site-packages/matplotlib-1.3.1-py2.7-linux-x86_64.egg/matplotlib/axes.py", line 461, in __init__
    self.cla()
  File "/usr/lib/python2.7/dist-packages/mpl_toolkits/mplot3d/axes3d.py", line 1051, in cla
    Axes.cla(self)
  File "/usr/local/sage/sage-6.5/local/lib/python2.7/site-packages/matplotlib-1.3.1-py2.7-linux-x86_64.egg/matplotlib/axes.py", line 922, in cla
    self.grid(self._gridOn)
  File "/usr/lib/python2.7/dist-packages/mpl_toolkits/mplot3d/axes3d.py", line 1262, in grid
    self._draw_grid = cbook._string_to_bool(b)
AttributeError: 'module' object has no attribute '_string_to_bool'
```

**Comment by haraldschilly** [comment]
I can confirm this, no idea how that did break. I'm certain that this did work some time ago and another test code is this:
```
import matplotlib.pyplot as plt
fig = plt.figure()
ax = fig.gca(projection = '3d')
```

**Comment by williamstein** [comment]
This will (I think) be fixed when I upgrade SMC to sage-6.7, which I intend
to do as soon as it is released (so very likely tomorrow):

salvus@compute2-us:/projects/sage/sage-6.7.beta5$ ./sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath Version 6.7.rc0, Release Date: 2015-05-14                 │
│ Type "notebook()" for the browser-based notebook interface.        │
│ Type "help()" for help.                                            │
└────────────────────────────────────────────────────────────────────┘
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Warning: this is a prerelease version, and it may be unstable.     ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

sage:
sage: import matplotlib.pyplot as plt
sage: from mpl_toolkits.mplot3d import Axes3D
sage: fig = plt.figure()
sage: ax = fig.add_subplot(111, projection='3d')
sage: fig.savefig('test.png')
sage: import matplotlib.pyplot as plt
sage: fig = plt.figure()
sage: ax = fig.gca(projection = '3d')

On Sun, May 17, 2015 at 2:25 AM, Harald Schilly <notifications@github.com>
wrote:

> I can confirm this, no idea how that did break. I'm certain that this did
> work some time ago and another test code is this:
>
> import matplotlib.pyplot as plt
> fig = plt.figure()
> ax = fig.gca(projection = '3d')
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sagemath/cloud/issues/196#issuecomment-102771600>.
>



-- 
William (http://wstein.org)

**Comment by haraldschilly** [comment]
this works again (a direct fix to the matplotlib library was necessary)
